### PR TITLE
le_tc/kernel : Modification in Kernel Testcases

### PR DESCRIPTION
**Pull request by bipin4321**
file:
itc_environ.c:commented snprintf which was causing corruption.
Testcase:-
 itc_environ_getenv_n

itc_libc_pthread.c:Removed Negative Testcases(As Per Posix
                   API,the return case is not defined)
Testcase:-
 itc_libc_pthread_pthread_rwlock_destroy_n_redestroy[Deleted]
 itc_libc_pthread_pthread_rwlock_trywrlock_n_after_destroy[Deleted]
 itc_libc_pthread_pthread_unlock_n_after_destroy[Deleted]
 itc_libc_pthread_pthread_rwlock_tryrdlock_n_after_destory[Deleted]
 itc_libc_pthread_pthread_condattr_destroy_n_redestroy[Deleted]
 itc_libc_pthread_pthread_attr_destroy_n_redestroy[Deleted]

itc_libc_semaphore.c:Modified return statement  as per the
                    implementation of sem_getprotocol in TizenRT.
Testcase:-
itc_libc_semaphore_getprotocol_n_no_sem_init

itc_libc_spwan.c:modification and deletion of few testcases
1)Following Testcase has been deleted,As there return is undefined as
     per Posix defination.
     itc_libc_spawn_posix_spawn_file_actions_destroy_p_redestroy [Deleted]
     itc_libc_spawn_file_actions_adddup2_n_after_destroy [Deleted]
     itc_libc_spawn_file_actions_addopen_n_after_destroy  [Deleted]
2)itc_libc_spawn_posix_spawnattr_setgetflags_n,testcase
   mofified as per the implementation in TizenRT.
3)itc_libc_spawn_posix_spawnattr_setgetschedparam_n,testcase
  modified as per the implementation in  TizenRT.
4)itc_libc_spawn_posix_spawnattr_setgetschedpolicy_n,testcase
  modified as per the  implementation in TizenRT.
5)itc_libc_spawn_posix_spawnattr_setgetsigmask_n,testcase
   modified as per the implementation  in TizenRT.
6)itc_libc_spawn_task_spawnattr_setgetstacksize_n,testcase
  modified as per the implementation in TizenRT.

itc_pthread.c:Deleted testcases as per the Posix API,the return case
is  undefined in  scenarios after destroy.
Testcases:-
itc_pthread_pthread_barrier_destroy_n_redestory[Deleted]
itc_pthread_pthread_cond_destroy_n_redestroy[Deleted]
itc_pthread_pthread_cond_broadcast_n_after_destroy[Deleted]
itc_pthread_mutex_lock_n_after_mutex_destroy[Deleted]
itc_pthread_mutex_destroy_n_redestroy[Deleted]

Signed-off-by: bipinkumar <bipin.k@samsung.com>

### CODE REVIEW

**Comment by bipin4321** [comment]
@samegautam ,Please review

File: apps/examples/testcase/le_tc/kernel/itc_environ.c [hunk 1]
```diff
@@ -89,7 +89,7 @@ static void itc_environ_getenv_n(void)
 
 	memset(var_buf, 0, BUFF_LEN);
 	snprintf(var_buf, BUFF_LEN, "%s%d=%s%d", VAR_STR, 0, VALUE_STR, 0);
-	snprintf(var_buf + strlen(var_buf) + 1, BUFF_LEN, "%s%d", VAR_STR, 1);
+	//snprintf(var_buf + strlen(var_buf) + 1, BUFF_LEN, "%s%d", VAR_STR, 1);
```
Comment (samegautam) [inline apps/examples/testcase/le_tc/kernel/itc_environ.c]: Please deleted the commented line

**Comment by seinfra** [comment]
Code Rule Check Result:
apps/examples/testcase/le_tc/kernel/itc_libc_spawn.c:114: ERROR: [SPC_M_OPR] space required after that ',' (ctx:VxV)
apps/examples/testcase/le_tc/kernel/itc_libc_spawn.c:115: ERROR: [SPC_M_OPR] space required after that ',' (ctx:VxV)

**Comment by seinfra** [comment]
Code Rule Check OK.

File: apps/examples/testcase/le_tc/kernel/itc_libc_semaphore.c [hunk 2]
```diff
@@ -49,8 +49,7 @@ static void itc_libc_semaphore_getprotocol_n_no_sem_init(void)
 
 	// sem is not initialized
 	ret_chk = sem_getprotocol(&sem, &protocol);
-	TC_ASSERT_EQ("sem_getprotocol", ret_chk, ERROR);
-	TC_ASSERT_EQ("sem_getprotocol", get_errno(), EINVAL);
+	TC_ASSERT_EQ("sem_getprotocol", SEM_PRIO_INHERIT, protocol);
```
Comment (jeongchanKim) [inline apps/examples/testcase/le_tc/kernel/itc_libc_semaphore.c]: SEM_PRIO_INHERIT is only when CONFIG_PRIORITY_INHERITANCE is enabled.

**Comment by girishks2000** [comment]
The justification of POSIX API compatibility is not true for test case removal. I checked pthread_* APIs, they do return the status. In TizenRT, it is EINVAL or EBUSY.  Need correct justification for tc removal. What is the exact problem, can it not be fixed. why removal is the only solution?

File: apps/examples/testcase/le_tc/kernel/itc_libc_semaphore.c (hunk 2 above)
Comment (bipin4321) [reply apps/examples/testcase/le_tc/kernel/itc_libc_semaphore.c]: This comment has been fixed,as per the comment.Further modification required in sem_getprotocol defination for proper handling  of the ERROR scenario.

File: apps/examples/testcase/le_tc/kernel/itc_environ.c (hunk 1 above)
Comment (bipin4321) [reply apps/examples/testcase/le_tc/kernel/itc_environ.c]: This comment has been fixed.

**Comment by seinfra** [comment]
Code Rule Check OK.

**Comment by seinfra** [comment]
Code Rule Check OK.

**Comment by bipin4321** [comment]
@jeongchanKim  @sunghan-chang ,please review

**Comment by seinfra** [comment]
Code Rule Check OK.

**Comment by sunghan-chang** [comment]
@bipin4321 Could you confirm that there is no failure on itc after merging this pull request?

**Comment by bipin4321** [comment]
@sunghan-chang : There are no failures,all the required changes has been done.

**Pull request merged**